Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -534,18 +534,24 @@ void RegisterToolbar()
 		void RemovePage(Page page)
 		{
 			IVisualElementRenderer rendererToRemove = Android.Platform.GetRenderer(page);
-			var containerToRemove = (PageContainer)rendererToRemove?.View.Parent;
 
-			// Also remove this page from the fragmentStack
-			FilterPageFragment(page);
-
-			containerToRemove.RemoveFromParent();
 			if (rendererToRemove != null)
 			{
-				rendererToRemove.View.RemoveFromParent();
-				rendererToRemove.Dispose();
+				var containerToRemove = (PageContainer)rendererToRemove.View.Parent;
+
+				rendererToRemove.View?.RemoveFromParent();
+
+				rendererToRemove?.Dispose();
+
+				if ((int)Build.VERSION.SdkInt < 25)
```

Comment:
This needs some kind of comment so we remember in the future why this API check is happening (especially since it might just be an Android bug and we might be able to remove it once the bug is fixed).